scripts/bootloader/hash.py

```diff
@@ -8,22 +8,26 @@
 import hashlib
 import sys
 import argparse
+from intelhex import IntelHex
 
 
 def parse_args():
     parser = argparse.ArgumentParser(
         description="Hash data from stdin or file.",
         formatter_class=argparse.RawDescriptionHelpFormatter)
 
-    parser.add_argument("--infile", "-i", "--in", "-in", required=False,
-                        type=argparse.FileType('rb'), default=sys.stdin.buffer,
+    parser.add_argument("--infile", "-i", "--in", "-in", required=True,
                         help="Hash the contents of the specified file instead of stdin.")
-
-    args = parser.parse_args()
-
-    return args
+    return parser.parse_args()
 
 
 if __name__ == "__main__":
     args = parse_args()
-    sys.stdout.buffer.write(hashlib.sha256(args.infile.read()).digest())
+
+    if args.infile.endswith('.hex'):
+        ih = IntelHex()
+        ih.loadhex(args.infile)
+        to_hash = ih.tobinstr()
```

Comment:
It might be worth giving an error if the hex file is not contiguous

Reply:
Hm, I'm not sure about that. The hash is still valid, and this might be used for something else then what it is currently used for, so it could be useful anyway.

Reply:
I think if the hex file has gaps, they are filled with padding bytes, which might not be what the user is expecting.